### DISCUSSION TITLE
How to connect Socket.IO client using HTTPS frontend to Socket.IO server running on HTTP ?

### DISCUSSION THREAD

**Discussion by Santheepkumar** [socketio/socket.io#5042] [Q&A] [answered]
Describe the bug
I am using Vercel for the frontend with HTTPS/SSL. My backend is on AWS Lightsail, proxied with Nginx to an Express.js server. In this case, my backend is HTTP, and my frontend is HTTPS.All my other APIs are working, but when I use a Socket.io server on Express, Socket.io demands HTTPS on both sides.I need some way to connect an HTTPS frontend to an HTTP Socket.io backend.I tried using a self-signed certificate on the backend, but that also caused the same issue.I am using a public IP to connect to my backend testing server. It is not possible to obtain an SSL certificate for an IP address from Let's Encrypt. There are some paid SSL providers, but I think this can be fixed using Socket.io in some way.Any ways to fix this?

To Reproduce

Here is my server config:

Socket.IO server version: x4.7.5

Server

import http from "http"
import express from "express"
import { Server } from "socket.io";

const app = express()
const server = http.createServer(app)
const io = new Server(server, {
      cors: ["https://myfrontendurl.vercel.app"]
});

io.on("connection", (socket) => {
  console.log(`connect ${socket.id}`);

  socket.on("disconnect", () => {
    console.log(`disconnect ${socket.id}`);
  });
});

server.listen(3000) // using nginx I am proxing to this port.
Socket.IO client version: x4.7.5

Client

import { io } from "socket.io-client";

const socket = io("https://myserverip/", {});

socket.on("connect", () => {
  console.log(`connect ${socket.id}`);
});

socket.on("disconnect", () => {
  console.log("disconnect");
});
Expected behavior
A Socket.io client must connect via HTTPS to a Socket.io server using HTTP in some ways.

Platform:

Device: Any Browser
OS: Any OS

**Comment by darrachequesne** [comment, marked as answer]
Hi! Unfortunately, I don't think this is possible, as the browser won't allow a HTTPS page call a resource over plain HTTP.

See also: https://developer.mozilla.org/en-US/docs/Web/Security/Mixed_content

**Reply by Santheepkumar** [reply]
Hey @darrachequesne, Thank you for assistance.
